### ISSUE TITLE
Navigation menu block: URL & permalink structure

### ISSUE THREAD

**Issue by sarahmonster** [WordPress/gutenberg#17204]
From #16821:

> URL and permalink structure

> First iteration should offer a link in sidebar to the permalink page.
> Any changes in the settings should reflect in the Nav block.

I'm not entirely clear on what we're asking for here, so I'd appreciate some clarification (cc @mtias @mapk @karmatosed). I'm going to do my best to guess what we're looking for for each item though, so we can move forward.

> First iteration should offer a link in sidebar to the permalink page.

I _think_ this means that we want to show a link to the page in the sidebar when the menu item is selected, like this:

<img width="319" alt="Screenshot 2019-08-26 at 14 05 34" src="https://user-images.githubusercontent.com/376315/63711967-9bc33300-c80a-11e9-8d07-d6c9345b7e20.png">

Is my understanding here correct, or are we looking for something different? It probably makes more sense to offer a link to the current page right next to the control for editing that link, so that users can contextualise the link better, but I'd expect to be able to edit that link in the block toolbar or in an inline way, rather than in the sidebar. (We might get more clarity here from #17116 once we have clear direction there.)

> Any changes in the settings should reflect in the Nav block.

I _think_ this just means "if I update my page URL, the link to that page in my menu also changes". If that's the case, I'm 100% in agreement there, and that's what I expected would happen. I'm not sure if we need any design revisions or visual mockups to show this though.

A related question is "what happens if I change the _title of the page_", in which case I'd suggest that we follow this flow:

![Flow showing choices if menu item source reference page is edited, and effect on menu item](https://user-images.githubusercontent.com/376315/63718625-fb283f80-c818-11e9-96b7-743c3a716baf.png)

Basically, if the user changes the _url_ of a page, we'd always update the link in the menu, so that it points to the intended page. If they've changed the menu item label so it no longer matches the page title (so I may change a page with a title of "Homepage" and give it the label "Home"), the edited label stays. If they haven't edited the label, the label updates to match the new page title.

**Comment by noisysocks** [comment]
Closing this for now as it's not totally clear what is needed. https://github.com/WordPress/gutenberg/issues/17544 tracks the effort towards a v1 of the Navigation block.
